src/libraries/System.Drawing.Common/src/System/Drawing/Font.cs

```diff
@@ -291,6 +291,32 @@ public unsafe void ToLogFont(object logFont, Graphics graphics)
             }
         }
 
+#if NET7_0_OR_GREATER
+        /// <summary>
+        /// Update the given LOGFONT with data from given graphics.
+        /// </summary>
+        /// <param name="logFont">A structure to populate with LOGFONT data.</param>
+        /// <param name="graphics">Graphics object which provide information about font.</param>
+        public unsafe void ToLogFont<T>(ref T logFont, Graphics graphics)
+            where T: struct, unmanaged
+        {
+            ArgumentNullException.ThrowIfNull(logFont);
+
+            int nativeSize = sizeof(Interop.User32.LOGFONT);
+            if (Marshal.SizeOf<T>() != nativeSize)
```

Comment:
```suggestion
            if (sizeof(T) != nativeSize)
```
It is not correct to use `Marshal.SizeOf` here